cot/benches/bench_utils.rs

```diff
@@ -0,0 +1,243 @@
+    name: &'static str,
+    #[allow(clippy::allow_attributes, dead_code)]
+    criterion: &'a mut Criterion,
+    method: reqwest::Method,
+    path: Option<&'static str>,
+    body: Option<String>,
+    content_type: Option<&'static str>,
+    expected_status_code: reqwest::StatusCode,
+    requests_per_iteration: u64,
+}
+
+macro_rules! builder_method {
+    ($name:ident, $ty:ty) => {
+        #[must_use]
+        pub(crate) fn $name(mut self, $name: $ty) -> Self {
+            self.$name = $name;
+            self
+        }
+    };
+    ($name:ident, $ty:ty, opt) => {
+        #[must_use]
+        pub(crate) fn $name(mut self, $name: $ty) -> Self {
+            self.$name = Some($name);
+            self
+        }
+    };
+}
+
+#[allow(clippy::allow_attributes, dead_code)]
+pub(crate) fn bench<'a>(
+    criterion: &'a mut Criterion,
+    name: &'static str,
+) -> ProjectBenchmarkBuilder<'a> {
+    ProjectBenchmarkBuilder::new(criterion, name)
+}
+
+#[allow(clippy::allow_attributes, dead_code)]
+impl<'a> ProjectBenchmarkBuilder<'a> {
+    pub(crate) fn new(criterion: &'a mut Criterion, name: &'static str) -> Self {
+        Self {
+            name,
+            criterion,
+            method: reqwest::Method::GET,
+            path: None,
+            body: None,
+            content_type: None,
+            expected_status_code: reqwest::StatusCode::OK,
+            requests_per_iteration: 50,
+        }
+    }
+
+    builder_method!(path, &'static str, opt);
+    builder_method!(body, String, opt);
+    builder_method!(content_type, &'static str, opt);
+    builder_method!(requests_per_iteration, u64);
+    builder_method!(method, reqwest::Method);
+    builder_method!(expected_status_code, reqwest::StatusCode);
+
+    pub(crate) fn plain_body<T: ToString>(mut self, body: &T) -> Self {
+        self.body = Some(body.to_string());
+        self.content_type = Some("text/plain");
+        self
+    }
+
+    pub(crate) fn json_body<T: serde::Serialize>(mut self, body: &T) -> Self {
+        self.body = Some(serde_json::to_string(body).expect("Failed to serialize JSON"));
+        self.content_type = Some("application/json");
+        self
+    }
+
+    pub(crate) fn form_body<T: serde::Serialize>(mut self, body: &T) -> Self {
+        self.body = Some(serde_urlencoded::to_string(body).expect("Failed to serialize form"));
+        self.content_type = Some("application/x-www-form-urlencoded");
+        self
+    }
+
+    pub(crate) fn run_with_router<F>(self, f: F)
+    where
+        F: FnOnce() -> Router,
+    {
+        let project = SimpleCotProject::new(self.name, f());
+        self.run_with_project(project);
+    }
+
+    pub(crate) fn run_with_project<P>(self, project: P)
+    where
+        P: Project + Send + 'static,
+    {
+        let (tx, rx) = std::sync::mpsc::sync_channel(1);
+        let _server_handle = std::thread::spawn(move || {
+            let runtime = tokio::runtime::Builder::new_current_thread()
```

Comment:
Why are we using `current_thread` tokio runtime? This is unrepresentative of real use cases, where we run the server on a multi-threaded runtime.